test/setCommand_test.js

```diff
@@ -18,48 +18,47 @@ describe('The set command', () => {
   beforeEach(() => {
     dataLayer = [];
     dlh = new DataLayerHelper(dataLayer);
-    commandAPI('set', 'bar', 'foo');
-    targetModel = Object.assign({}, dlh.abstractModelInterface_);
+    targetModel = {};
   });
 
   describe('with 2 argument format', () => {
     it('updates model without corrupting it.', () => {
       commandAPI('set', 'foo', 'bar');
       targetModel['foo'] = 'bar';
 
-      expect(dlh.abstractModelInterface_).toEqual(targetModel);
+      expect(dlh.get('')).toEqual(targetModel);
```

Comment:
I'm not sure I agree with the empty string returning the entire model. It's not very readable / intuitive.

Perhaps having the entire model returned when there are no arguments would look a bit better?
```
dlh.get() // returns entire model as no specific key has been provided
dlh.get('') // returns what is stored in empty key
dlh.get('example') // returns what is stored in example key
```

I support the idea of returning the entire model. A nifty side effect is that it allows people to define and use multiple top level keys at the same time via object destructuring:

```
// if a key doesn't exist in the model, it'll be set to undefined
const { client_id, page_path } = dlh.get();
```

Reply:
Edit: I just realized we need to be careful about returning the entire model. Objects are passed by reference in JavaScript, so by returning the model via reference we allow people to directly mutate the entire internal model.

It seems we already return objects in the model by reference and therefore allow the user to directly mutate objects within the model, but allowing this same behavior for the model itself could be allowing the users to shoot themselves in the foot easily... Since we already support returning objects within the model by reference, we could create a shallow copy of the model with those references intact and return that instead?

I'm open to ideas but we might want to get approval via a reviewed design doc before we make changes to get.

Reply:
Thanks for the suggestions, I got rid of the changes to get, we can do it in the future maybe.